src/main/java/io/stargate/sgv2/jsonapi/service/resolver/sort/TableMemorySortClauseResolver.java

```diff
@@ -70,6 +77,22 @@ public WithWarnings<RowSorter> resolve(
       throw new IllegalStateException("Cannot do table vector sorts in memory");
     }
 
+    // because pagination currently only comes from the CQL driver and in memory sort reads all the
+    // data
+    // we cannot support pagination with in memory sorting.
+    if (!cqlPageState.isEmpty()) {
+      var apiTableDef = commandContext.schemaObject().apiTableDef();
+
+      throw SortException.Code.UNSUPPORTED_PAGINATION_WITH_IN_MEMORY_SORTING.get(
+          errVars(
+              commandContext.schemaObject(),
+              map -> {
+                map.put(
+                    "partitionSorting", errFmtApiColumnDef(apiTableDef.clusteringKeys().values()));
+                map.put("sortColumns", errFmtCqlIdentifier(sortClause.sortColumnIdentifiers()));
```

Comment:
We want the error message to be stable, so we can check it in tests. 

So we need to sort the list of `sortClause.sortColumnIdentifiers()` , use the comparator in the CqlIdentifierUtil class

Reply:
code is correct, this is the sort order from the user we should keep that.